phantombuster/aws-js-s3-explorer
# LG-1308 fix: s3 objects are now paginated correctly

## 📝 What is this PR

**Describe what your PR addresses, could be a bug or a feature request**

-   [Specs](https://phantombuster.atlassian.net/browse/LG-1308?atlOrigin=eyJpIjoiYjcyMmQ3Njc4OTkzNDM5NmE3ZTFkMTNiYTJlY2ZiMWQiLCJwIjoiaiJ9)

## 🏭 How does it work

For some reason, the `changeViewPrefix` method was called even though the prefix did not really change.
This ultimately triggers a refresh using the `$scope.refresh` function which ends up setting `$scope.stop`.

The `listobjectscb` that is called after fetching the objects from s3 checks if `$scope.stop` is set to `true`. (line 567 of explorer.js)
When that's the case, the remaining objects are not fetch even when `IsTruncated` is returned by s3.

## 🧪 QA Process

- Use the process described in the [README](https://github.com/phantombuster/aws-js-s3-explorer/tree/prod) to run the s3 explorer locally.
- Run the front with the staging config and navigate to this Phantom: https://localhost:1234/1111111122222222/phantoms/4800951370520884/console (log with user+test.pb2.back@mg.phantombuster.com)
- Make sure more than 1k files are retrieved (should be around 2106 entries)
- Make sure there are no regressions: check navigating pages, downloading, uploading, navigating through all org files...)
- Make sure it still works with agents that have less than 1000 files

![image](https://user-images.githubusercontent.com/8069826/212694698-31b76aa7-71af-4622-8321-0d906ce8db43.png)


## 👌 To check

-   [ ] There's no regression
-   [ ] Coding guidelines are followed ([Contributing](https://github.com/phantombuster/engineering-docs/blob/master/contributing.md), [Testing](https://github.com/phantombuster/engineering-docs/blob/master/testing.md))

## 🚀 What to do after deployment

**Who should be notified after the deployment of this PR? What should be written in the changelog on Slack?**
